ledger/sandbox/src/main/scala/com/digitalasset/platform/apiserver/services/ApiCommandService.scala

```diff
@@ -74,92 +75,86 @@ final class ApiCommandService private (
   }
 
   @SuppressWarnings(Array("org.wartremover.warts.Any"))
-  private def submitAndWaitInternal(request: SubmitAndWaitRequest): Future[Completion] = {
-
-    val appId = request.getCommands.applicationId
-    val submitter = TrackerMap.Key(application = appId, party = request.getCommands.party)
-
-    if (running) {
-      submissionTracker
-        .track(submitter, request) {
-          for {
-            trackingFlow <- {
-              lowLevelCommandServiceAccess match {
-                case LowLevelCommandServiceAccess.LocalServices(
-                    submissionFlow,
-                    getCompletionSource,
-                    getCompletionEnd,
-                    _,
-                    _) =>
-                  for {
-                    ledgerEnd <- getCompletionEnd().map(_.getOffset)
-                  } yield {
-                    val tracker =
-                      CommandTrackerFlow[Promise[Completion], NotUsed](
-                        submissionFlow,
-                        offset =>
-                          getCompletionSource(
-                            CompletionStreamRequest(
-                              configuration.ledgerId.unwrap,
-                              appId,
-                              List(submitter.party),
-                              Some(offset)))
-                            .mapConcat(CommandCompletionSource.toStreamElements),
-                        ledgerEnd
-                      )
-
-                    if (configuration.limitMaxCommandsInFlight)
-                      MaxInFlight(configuration.maxCommandsInFlight).joinMat(tracker)(Keep.right)
-                    else tracker
-                  }
+  private def submitAndWaitInternal(request: SubmitAndWaitRequest): Future[Completion] =
+    withEnrichedLoggingContext(
+      logging.commandId(request.getCommands.commandId),
+      logging.party(request.getCommands.party)) { implicit logCtx =>
```

Comment:
Makes subsequent diffs nicer if we use trailing commas.

```suggestion
      logging.party(request.getCommands.party),
    ) { implicit logCtx =>
```

There's a few other places in the diff too, which you're welcome to address or not as you like.

Reply:
I get a compilation error if I try to do this, are you sure this is valid syntax for varargs?

Reply:
Addressed by https://github.com/digital-asset/daml/pull/4205/commits/c887bf168c6fd7d8331aeedd61e5c284d1e47ae8

Reply:
> I get a compilation error if I try to do this, are you sure this is valid syntax for varargs?

TIL trailing commas are valid only before a trailing newline. Thanks for clarifying this offline.